### DISCUSSION TITLE
ESP32-S3 onboard ws2812 led as Ledlink.

### DISCUSSION THREAD

**Discussion by valdus2** [arendst/Tasmota#21800] [Q&A]
As the title says, is it possible to use the onboard ws2812 led in an ESP32S3 as ledlink?
I tried changing it in template from ws2812 to ledlink but nothing happened.

Thank you.

**Comment by Jason2866** [comment]
No, not possible. ws2812 needs the Neopixel Lib which is not in all build variants included.
Even when the code is not there to use a ws2812 with ledlink

**Comment by HofeBY** [comment]
Maybe you find ideas on this site how to solve it.
https://www.reichelt.de/magazin/projekte/die-433-mhz-tasmota-alarmanlage/
search documentation about Wifi#Connected

**Comment by valdus2** [comment]
@Jason2866 Maybe I haven't understand what you mean exactly or @both maybe I am misunderstood...

1) Installing Tasmota and choosing autoconfig with DevKitC-1:

![2](https://github.com/user-attachments/assets/6e171bf5-757e-4370-aebe-fb28df4e3d3f)

2) This is the default template with gpio48 declared as the onboard ws2812 and not at my disposal:

![3](https://github.com/user-attachments/assets/7ca7b3b5-5ced-4f33-91c1-375069d3c1f8)


3) And the webUI correctly showing and controlling the onboard ws2812 led:

![1](https://github.com/user-attachments/assets/0b4fbc38-3277-4e57-a243-f9a122e49101)

 
 
.
.
.
 
_**And now comes the question if I can use it as ledlink and how?**_

**Reply by sfromis** [reply]
Jason2866 already clearly answered that question. "No, not possible", and "Even when the code is not there to use a ws2812 with ledlink". Not a feature.

But as hinted by HofeBY, nothing would stop you from creating something similar, 100% by yourself, keeping track of status, and tickle the neopixel.

**Comment by valdus2** [comment]
Lliterally, I didn't understand what @Jason2866 meant....
I thought he was talking about the code for ws2812 that was not there, not for using it as Ledlink...

So, thank you all.
